python_modules/libraries/dagster-sigma/dagster_sigma/resource.py

```diff
@@ -0,0 +1,62 @@
+from enum import Enum
+from typing import Any, Dict, Optional
+
+import requests
+from dagster import ConfigurableResource
+from pydantic import Field, PrivateAttr
+
+
+class SigmaCloudType(Enum):
+    AWS_US = "https://aws-api.sigmacomputing.com"
+    AWS_CANADA = "https://api.ca.aws.sigmacomputing.com"
+    AWS_EUROPE = "https://api.eu.aws.sigmacomputing.com"
+    AWS_UK = "https://api.uk.aws.sigmacomputing.com"
+    AZURE_US = "https://api.us.azure.sigmacomputing.com"
+    GCP = "https://api.sigmacomputing.com"
+
+
+class SigmaOrganization(ConfigurableResource):
+    """Represents a workspace in PowerBI and provides utilities
+    to interact with the PowerBI API.
+    """
+
+    cloud_type: SigmaCloudType = Field(
+        ...,
+        description="The cloud type for your Sigma organization, found under the Administration -> Account -> Site settings.",
+    )
```

Comment:
Might be easier to just make this field a string and rename it to `base_url`. We can still validate the URL on resource initialization.

That way, users don't need to figure out the indirection of `SigmaCloudType` and the `Enum` class.

We should also probably link out to https://help.sigmacomputing.com/docs/region-warehouse-and-feature-support#supported-cloud-platforms-and-regions

Reply:
I think it's helpful to have as an enum just because the URLs are only available in docs and not what's shown to the user - in the settings page it just tells you what cloud you have (e.g. in this case just AWS, which is AWS US).

<img width="350" alt="Screenshot 2024-09-16 at 10 20 00 AM" src="https://github.com/user-attachments/assets/2e11f479-3708-4ae2-b481-c86d6a8b7c40">

Reply:
Added docs link